### ISSUE TITLE
Configuration to collapse groups by default

### ISSUE THREAD

**Issue by tylla** [Ylianst/MeshCentral#5361]
We have lots of machines in lots of groups and we are frequently using MeshCentral from some other computer (meaning, not our regularly used work desks, but some random users machine).
Every time we have to scroll trough the immense list of machines or hit the "Collapse All" button.

It would be wonderful to have a configuration option to have the list collapsed each time a new browser comes at play.
The new users, and users with few hosts wouldn't even know about this, but users who have lots of machines and are jumping browsers constantly would have the possibility to have all the groups collapsed and be able to choose the one that they are interested in.
Naturally the current method of saving the expanded/collapsed state to a cookie on the current browser would be unaffected by this option, this would only kick in as the default case.

Thank you for this nice software and feel free to ask if anything I wrote was not clear.

**Comment by si458** [comment]
Would a remember feature be better?
So if u collapsed all the groups and refreshed the page, it still collapsed?
And same if some are collapsed abd some arent, then keep those collapsed and others uncollapsed?

**Comment by si458** [comment]
Scratch previous comment, just tried on my device

If I collapse groups, then refresh or logout, Login, they stay collapsed

I'm guessing u are referring to logging in and out on multiple devices?

**Comment by Ylianst** [comment]
There is a collapse all / expand all option here. Let us know if this works:

![image](https://github.com/Ylianst/MeshCentral/assets/1319013/85487fa9-0787-446c-b2d7-4817479bb48e)

**Comment by tylla** [comment]
@Ylianst thank you for the reply, I am aware of that nice function, and I'm using it as well (albeit I have some other improvement ideas for it as mentioned in #5360).

Rather this feature request is about the default behavior, when someone logs in to the site from a new browser (so there is no preferences cookie saved).
I would like to set the setting to collapsed, so each time a new browser enters the playground, it gets all groups collapsed.

This should be a user setting, so that those users who get only a few hosts listed will keep it on its default (expanded), but those with tens-hundreds of hosts can set their setting to collapsed, so they get a nice overview of their groups at first login.

I hope this clears up a little bit. If not please ask, I'm more than happy to elaborate!

**Comment by si458** [comment]
@tylla ahh... I understand what you mean! Just tried it myself
If I login in my normal browser then collapse the groups and logout/login it stays that way,
BUT if I then login in private mode, they are all expanded again rather than collapsed,
You want the collapse feature to be stored on the server side, then wherever you login, it's remembers where u left off (as such)

**Comment by tylla** [comment]
This would be a nice extra feature, please consider reopening it, maybe in the future would get some attention.

**Comment by tylla** [comment]
Off: no not stale, just sleeping... :)

**Comment by si458** [comment]
just relooking into this, is this still needed?
currently it stores the `collapse` state of the groups in the browsers `localSession` so it doesnt copy to other machines/browsers
did you want this to copy to other browsers?
or just the original request to have it collapsed by default if the user doesnt have a `collapse` state set?
or both? (haha)

**Comment by tylla** [comment]
@si458 Nah, I don't want to sound too greedy... just my original idea. :-)
**To be able to set the default collapse behavior on new browser sessions.**
As you wrote previously (I know, it's been a long time, I had to read a few paragraphs back as well. :):
> [@tylla](https://github.com/tylla) ahh... I understand what you mean! Just tried it myself If I login in my normal browser then collapse the groups and logout/login it stays that way, BUT if I then login in private mode, they are all expanded again rather than collapsed, You want the collapse feature to be stored on the server side, then wherever you login, it's remembers where u left off (as such)

The "store the collapse setting on server side" is a good enough solution.

An even better one would be a global setting e.g. "Machine list Collapse all default state" which would say what to do on a new browser session (either new machine, new browser or private browsing). This setting should be "Expanded" as a factory default, so all those people using MC for some time don't get a nasty surprise. But if set to "Collapsed" then every new session would get a collapsed machine list as a default. And naturally every recurring session would load it's last collapsed/expanded setting from the `localSession` as the user left it according to the current functionality.

**Comment by si458** [comment]
@tylla thanks for the reply, ok so ill do the following

1. when you `collapse/expand` it will save to `LocalSession` AND to MeshCentral for safe keeping
2. when you login elsewhere it will download your preferences and loads them into `LocalSession` and your session
(it already does this, i just need to add in `collapse` into it for save/load/etc)
2. add a new option under domains in config.json and any browser that logs in and DOESNT have `collapse` in `LocalSession` OR from the server (like a new fresh user OR something broke with the server and it didnt save your preferences)
it will use this value to `collapse/expand` what it needs to be shown as

**Comment by tylla** [comment]
If I understand you correctly, you would like to save the `collapse/expand` state of the device tree to the server?! (in parallel with the `LocalSession`)
I like the concept, and it makes sense for lots of users, but I can think of a few scenarios where this could cause major annoyances for some users.
E.g. I can imagine a situation where a user has multiple browsers with the same user login, but s/he tends to use a different part of the device tree on the different browsers. The central saving and downloading of the tree state would ruin this use case, as all his/her browsers would get synchronized automatically. If one would like such an otherwise nice feature, then this should be made optional, a'la "Sync device list tree state between sessions" option.
And bang here we go with another option. ;-)

**Comment by si458** [comment]
> E.g. I can imagine a situation where a user has multiple browsers with the same user login, but s/he tends to use a different part of the device tree on the different browsers. The central saving and downloading of the tree state would ruin this use case, as all his/her browsers would get synchronized automatically. If one would like such an otherwise nice feature, then this should be made optional, a'la "Sync device list tree state between sessions" option.

And i think this is the exact reason the `collapse` value isnt currently being sycnced to meshcentral and between devices but stayed locally in the Web browser instead!

**Comment by si458** [comment]
ok done and will be next release or tomorrows docker `master` release! 👍 https://github.com/Ylianst/MeshCentral/commit/2db9420715a82daebfec817880ec9e45023f9f9b

so you can add `"collapseGroups": true`  under the `domains` section
and it will check if its set to true and if the web browser didnt have any value before hand and if you havent got it set in memory too
then it will collapse it by default for you

**Comment by tylla** [comment]
Wow, thanks!
We will get around updating our MeshCentral and checking it out soon. Will report back.
